From 8319405758bdaed476053eacb8ec8a2bb12adfa0 Mon Sep 17 00:00:00 2001
From: Robert Chiras <robert.chiras@nxp.com>
Date: Fri, 2 Mar 2018 11:35:53 +0200
Subject: [PATCH 3420/5242] MLK-17650-2: arm64: dts: fsl-imx8mq-evk: Update
 adv7535

commit  db5ecdcfce28c6101490a39c2a2494d9d2fae626 from
https://source.codeaurora.org/external/imx/linux-imx.git

Since the ADV7535 address for DSI-CEC can be configured from DTS file,
update the ADV7535 specific files in order to remove the disabling
camera node and update the ADV7535 node to use a different i2c address.
Currently 0x3c was used, conflicting with camera. Now, program ADV7535
to use 0x3b for the DSI-CEC memory map.

Signed-off-by: Robert Chiras <robert.chiras@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts  |    5 +----
 .../dts/freescale/fsl-imx8mq-evk-dual-display.dts  |    5 +----
 .../dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts |    5 +----
 3 files changed, 3 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts
index 360bbb3..e049bab 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dcss-adv7535.dts
@@ -14,10 +14,6 @@
 
 #include "fsl-imx8mq-evk.dts"
 
-&ov5640_mipi {
-	status = "disabled";
-};
-
 &hdmi {
 	status = "disabled";
 };
@@ -39,6 +35,7 @@
 	adv_bridge: adv7535@3d {
 		compatible = "adi,adv7533";
 		reg = <0x3d>;
+		adi,addr-cec = <0x3b>;
 		adi,dsi-lanes = <4>;
 		status = "okay";
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display.dts
index 079e9f1..2c51269 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-dual-display.dts
@@ -14,10 +14,6 @@
 
 #include "fsl-imx8mq-evk.dts"
 
-&ov5640_mipi {
-	status = "disabled";
-};
-
 &lcdif {
 	status = "okay";
 
@@ -32,6 +28,7 @@
 	adv_bridge: adv7535@3d {
 		compatible = "adi,adv7533";
 		reg = <0x3d>;
+		adi,addr-cec = <0x3b>;
 		adi,dsi-lanes = <4>;
 		status = "okay";
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts
index b45f6be..204d696 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-evk-lcdif-adv7535.dts
@@ -24,10 +24,6 @@
 	status = "disabled";
 };
 
-&ov5640_mipi {
-	status = "disabled";
-};
-
 &dcss {
 	status = "disabled";
 };
@@ -50,6 +46,7 @@
 	adv_bridge: adv7535@3d {
 		compatible = "adi,adv7533";
 		reg = <0x3d>;
+		adi,addr-cec = <0x3b>;
 		adi,dsi-lanes = <4>;
 		status = "okay";
 
-- 
1.7.9.5

